java-client/src/test/java/org/opensearch/client/opensearch/experiments/ParsingTests.java

```diff
@@ -124,4 +136,22 @@ public void testIndexSettingsMappingParsing() {
       throw new JsonParsingException(je.getMessage() + " at " + je.getLocation(), je, je.getLocation());
     }
   }
+
+  private <T extends JsonpSerializable> T deserialize(String serializedValue, JsonpDeserializer<T> deserializer) {
+    var provider = JsonProvider.provider();
+    var parser = provider.createParser(new ByteArrayInputStream(serializedValue.getBytes(StandardCharsets.UTF_8)));
+
+    return deserializer.deserialize(parser, new JsonbJsonpMapper());
+  }
+
+  private String serialize(JsonpSerializable object) {
+    var baos = new ByteArrayOutputStream();
+    var provider = JsonProvider.provider();
+
+    var generator = provider.createGenerator(baos);
+    object.serialize(generator, new JsonbJsonpMapper());
+    generator.close();
+
+    return baos.toString();
+  }
```

Comment:
We should move this somewhere into a util class and reuse.

Reply:
Looks like it is `ModelTestCase`'s responsibility. Added a third commit which makes test code use its methods when dealing with jsons.